onebusaway-android/src/main/java/org/onebusaway/android/directions/realtime/RealtimeService.java

```diff
@@ -13,13 +14,258 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 package org.onebusaway.android.directions.realtime;
 
+import org.onebusaway.android.R;
+import org.onebusaway.android.directions.model.ItineraryDescription;
+import org.onebusaway.android.directions.tasks.TripRequest;
+import org.onebusaway.android.directions.util.OTPConstants;
+import org.onebusaway.android.directions.util.TripRequestBuilder;
 import org.opentripplanner.api.model.Itinerary;
+import org.opentripplanner.api.model.Leg;
+
+import android.app.Activity;
+import android.app.AlarmManager;
+import android.app.IntentService;
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.preference.PreferenceManager;
+import android.support.v4.app.NotificationCompat;
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+
+public class RealtimeService extends IntentService {
+
+    private static final String TAG = "RealtimeService";
+
+    public RealtimeService() {
+        super("RealtimeService");
+    }
+
+    /**
+     * Start realtime updates.
+     *
+     * @param source Activity from which updates are started
+     * @param bundle Bundle with selected itinerary/parameters
+     */
+    public static void start(Activity source, Bundle bundle) {
+        bundle.putSerializable(OTPConstants.NOTIFICATION_TARGET, source.getClass());
+        Intent intent = new Intent(OTPConstants.INTENT_START_CHECKS);
+        intent.putExtras(bundle);
+        source.sendBroadcast(intent);
+    }
+
+    @Override
+    public void onHandleIntent(Intent intent) {
+        Bundle bundle = intent.getExtras();
+
+        Itinerary itinerary = getItinerary(bundle);
+        Class<? extends Activity> activity = (Class<? extends  Activity>)
+                bundle.getSerializable(OTPConstants.NOTIFICATION_TARGET);
+
+        if (intent.getAction().equals(OTPConstants.INTENT_START_CHECKS)) {
+            disableListenForTripUpdates();
+            if (!possibleReschedule(bundle)) {
+                startRealtimeUpdates(bundle, itinerary);
+            }
+        } else if (intent.getAction().equals(OTPConstants.INTENT_CHECK_TRIP_TIME)) {
+            checkForItineraryChange(bundle, itinerary, activity);
+        }
+
+        RealtimeWakefulReceiver.completeWakefulIntent(intent);
+    }
+
+    // Depending on preferences / whether there is realtime info, start updates.
+    private void startRealtimeUpdates(Bundle params, Itinerary itinerary) {
+
+        Log.d(TAG, "Checking whether to start realtime updates.");
+
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
```

Comment:
I'd prefer to use our singleton method to get the preferences, as used elsewhere in the app:
`SharedPreferences prefs = Application.getPrefs();`